### PR TITLE
Beats macOS weekly pipeline

### PR DESCRIPTION
**Pull request by oakrizan**
## Proposed commit message
Added all available macOS steps to weekly pipeline.
Should be merged after https://github.com/elastic/beats/pull/41308

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates: https://github.com/elastic/ingest-dev/issues/3271

## Logs
Buildkite logs: https://buildkite.com/elastic/beats-macos-tests/builds/18
Tested changes in branch from original repo, not fork

### CODE REVIEW

**Comment by oakrizan** [comment]
This should be a short-time solution, as with AWS tests.
Related issue: https://github.com/elastic/ingest-dev/issues/4336

File: .buildkite/macos-tests-pipeline.yml [hunk 1]
```diff
@@ -14,14 +14,536 @@ env:
...
+  - group: "Auditbeat macOS Tests"
+    steps:
+    - label: ":mac: Auditbeat: macOS x86_64 Unit Tests"
+      command: |
+        set -euo pipefail
+        source .buildkite/scripts/install_macos_tools.sh
+        cd auditbeat
+        mage build unitTest
+      concurrency_group: "orka-concurrency-group"
+      concurrency: 10
```
Comment (pazone) [inline .buildkite/macos-tests-pipeline.yml:26]: Shouldn't we add `concurrency_method: eager` here?

**Review by pazone — APPROVED** [review]
LGTM

**Pull request merged**